src/libgit2/tag.c

```diff
@@ -540,13 +554,7 @@ int git_tag_name_is_valid(int *valid, const char *name)
 
 	GIT_ASSERT(valid);
 
-	*valid = 0;
-
-	/*
-	 * Discourage tag name starting with dash,
-	 * https://github.com/git/git/commit/4f0accd638b8d2
-	 */
-	if (!name || name[0] == '-')
+	if (!name || !tag_name_is_valid(name))
```

Comment:
```suggestion
	*valid = 0;

	if (!name || !tag_name_is_valid(name))
```

Reply:
Don't forget to set validity to false when we exit on `!name || !tag_name_is_valid(name)`